data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -93,7 +94,8 @@ protected DeleteFilter(
 
     this.posDeletes = posDeleteBuilder.build();
     this.eqDeletes = eqDeleteBuilder.build();
-    this.requiredSchema = fileProjection(tableSchema, requestedSchema, posDeletes, eqDeletes);
+    this.requiredSchema =
+        fileProjection(tableSchema, requestedSchema, posDeletes, eqDeletes, isBatchReading);
```

Comment:
One question I asked myself is whether it impacts the metadata column read? It seems not, but the method `DeleteFilter::fileProjection` seems a bit hard to read, we can refactor it later. It makes more sense to make it a static until method instead of instance method. It's a bit weird to pass the schema to delete filter, then get it back from the filter. This seems something we can improve on it as a follow up.